src/silo/zstdfasta/zstdfasta_table.cpp

```diff
@@ -0,0 +1,82 @@
+#include "silo/zstdfasta/zstdfasta_table.h"
+
+#include <fmt/format.h>
+#include <duckdb.hpp>
+
+#include "silo/common/fasta_reader.h"
+#include "silo/preprocessing/preprocessing_exception.h"
+#include "silo/zstdfasta/zstd_compressor.h"
+#include "silo/zstdfasta/zstdfasta_reader.h"
+
+namespace {
+void initializeTable(duckdb::Connection& connection, std::string table_name) {
+   auto return_value = connection.Query(fmt::format("DROP TABLE IF EXISTS {};", table_name));
+   if (return_value->HasError()) {
+      throw silo::preprocessing::PreprocessingException(return_value->ToString());
+   }
+   return_value = connection.Query(fmt::format(
+      "CREATE TABLE {} ("
+      "    key STRING,"
+      "    sequence BLOB"
+      ");",
+      table_name
+   ));
+   if (return_value->HasError()) {
+      throw silo::preprocessing::PreprocessingException(return_value->ToString());
```

Comment:
```suggestion
      throw silo::preprocessing::PreprocessingException(return_value->GetError());
   }
   return_value = connection.Query(fmt::format(
      "CREATE TABLE {} ("
      "    key STRING,"
      "    sequence BLOB"
      ");",
      table_name
   ));
   if (return_value->HasError()) {
      throw silo::preprocessing::PreprocessingException(return_value->GetError());
```
since we've been using that everywhere else, too. Or does `toString` also work?

I wonder wether it might make sense to extract a macro here? something like: `CHECK_QUERY(return_value, "error when executing this specific statement")` which results in something like
```
   if (return_value->HasError()) {
      throw silo::preprocessing::PreprocessingException("error when executing this specific statement" + ": " + return_value->GetError());
   }
```

Reply:
Part of the reason why I created the duckdb wrapper as preprocessing_database was to have a central logging location, but to get the abstractions there correct was somehow a little difficult, thus I made some compromises. I will try to refactor that again

Reply:
The `preprocessing_database` wrapper definitely makes sense. I think this is just some detail where we can remove duplication. Maybe it can also be a normal function (since macros in C++ are kinda nasty)?